cpp/modmesh/python/common.hpp

```diff
@@ -505,21 +505,43 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY Interpreter
 // Suppress the warning "greater visibility than the type of its field"
 #pragma GCC diagnostic ignored "-Wattributes"
 #endif
-class PyStdErrOutStreamRedirect
+class PythonStreamRedirect
 {
 
 public:
-    PyStdErrOutStreamRedirect();
-    std::string stdout_string();
-    std::string stderr_string();
-    ~PyStdErrOutStreamRedirect() noexcept(false);
+
+    PythonStreamRedirect(bool enabled)
+        : m_enabled(enabled)
+    {
+    }
+
+    std::string stdout_string() const;
+    std::string stderr_string() const;
+
+    PythonStreamRedirect & set_enabled(bool enabled)
+    {
+        m_enabled = enabled;
+        return *this;
+    }
+
+    bool is_enabled() const { return m_enabled; }
```

Comment:
it looks redundant to have is_enabled and is_disable?

Reply:
It is intentional.  Having the symmetric helpers encapsulates better.  When the system grows we will forget how it is implemented inside and wonder it's `is_enabled()` or `is_disabled()`.  Implementing both resolves the ambiguity.

An IDE with auto-completion helps, but we cannot assume everyone uses an IDE.